test/transaction_test.exs

```diff
@@ -98,6 +99,44 @@ defmodule TransactionTest do
     assert query("SELECT 42", []) == [[42]]
   end
 
+  @tag mode: :transaction
+  @tag disconnect_on_read_only_transaction_error: true
+  test "transaction read-only only error disconnects with prepare and execute", context do
+    Process.flag(:trap_exit, true)
+
+    assert transaction(fn conn  ->
+      P.query!(conn, "SET TRANSACTION READ ONLY", []).connection_id
+
+      {:ok, query} = P.prepare(conn, "query_1", "insert into uniques values (1);", [])
+
+      assert capture_log(fn ->
+        {:error, %Postgrex.Error{postgres: %{message: "cannot execute INSERT in a read-only transaction"}}} =
+          P.execute(conn, query, [])
+
+        pid = context[:pid]
+        assert_receive {:EXIT, ^pid, :killed}
```

Comment:
@fishcakez why killed? Are those exists going to count towards the supervisor max_restarts? If so, I can see this being a problem.

Reply:
To clarify, I thought that only stops counted as exits, and not disconnects.